src/NuGet.Configuration/PackageSource/PackageSource.cs

```diff
@@ -58,6 +59,30 @@ public bool IsHttp
             }
         }
 
+        /// <summary>
+        /// True if the source path is file based. Unc shares are not included.
+        /// </summary>
+        public bool IsLocal
+        {
+            get
+            {
+                if (!_isLocal.HasValue)
```

Comment:
This follows the same pattern as IsHttp. IsLocal will only be hit for packages.config restores.